pkg/keystone/cronjob.go

```diff
@@ -0,0 +1,125 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package keystone
+
+import (
+	keystonev1beta1 "github.com/openstack-k8s-operators/keystone-operator/api/v1beta1"
+	common "github.com/openstack-k8s-operators/lib-common/modules/common"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/affinity"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/env"
+
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// TrustFlushCommand -
+	TrustFlushCommand = "/usr/local/bin/kolla_set_configs && keystone-manage trust_flush"
+)
+
+// CronJob func
+func CronJob(
+	instance *keystonev1beta1.KeystoneAPI,
+	labels map[string]string,
+	annotations map[string]string,
+) *batchv1.CronJob {
+	runAsUser := int64(0)
+
+	args := []string{"-c"}
+	if instance.Spec.Debug.Service {
+		args = append(args, common.DebugCommand)
+	} else {
+		args = append(args, TrustFlushCommand)
+	}
+
+	envVars := map[string]env.Setter{}
+	envVars["KOLLA_CONFIG_FILE"] = env.SetValue(KollaConfig)
+	envVars["KOLLA_CONFIG_STRATEGY"] = env.SetValue("COPY_ALWAYS")
+
+	parallelism := int32(1)
+	completions := int32(1)
+
+	cronjob := &batchv1.CronJob{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ServiceName + "-cron",
+			Namespace: instance.Namespace,
+		},
+		Spec: batchv1.CronJobSpec{
+			Schedule:          "1 * * * *",
```

Comment:
I think that's a good value, just wondering if we ever need to expose the `schedule` for cronjobs? maybe not for this one, just a general thought.

Reply:
Yeah it makes sense to expose the schedule. Probably I can expose this in keystone-operator as well, so that users can tune frequency of the purge job.

In the past we provided an option to disable the cron job but I'm not sure that is required. We probably can omit it in operators until we hear real requirements.

Finally, I noticed I have to provide the interface to customize arguments of that command (eg ``--date``). I'll add that.

Reply:
I've added the parameter to expose the schedule and additional command arguments.